### ISSUE TITLE
`declarationDir` compiler option fails generation

### ISSUE THREAD

**Issue by igorpupkinable** [timocov/dts-bundle-generator#147]
I am trying to bundle type definitions, but get the following `error TS6053: File 'src/index.d.ts' not found.`
I run `yarn dts-bundle-generator --external-imports=react --out-file lib/index.d.ts src/index.ts` and it fails with this error.

My `src/index.ts` is just a simple import of reusable React components:
```
export {
  Body,
  Headline,
} from './Text';
```

`src/Text/index.ts` content is
```
export { default as Body } from './Body';
export { default as Headline } from './Headline';
```

`src/Text/Body.tsx` content is similar in both files:
```
import React, {
  FC,
} from 'react';
import classNames from 'classnames';

import { TextProps } from './TextProps';

import styles from './Text.css';

const Body: FC<TextProps> = ({
  ...
}) => (
  <span ... />
);

export default Body;
```

Why does it need `src/index.d.ts` at all?
I works fine if I run `tsc` to generate definition file for each ts/tsx file and use dts-bundle-generator later with
`yarn dts-bundle-generator --external-imports=react --out-file lib/index.d.ts lib/src/index.d.ts`

**Comment by timocov** [comment]
> Why does it need src/index.d.ts at all?

That's because the tool firstly compiles all sources to declaration files (in-memory) and then create a program for declaration files - this is how the tool works actually.

Also, what version of the tool you use? It seems that the issue is related to "outDir" folder (I noticed that source code is in `src` folder, but output for it is `lib/src`) but [the tool removes](https://github.com/timocov/dts-bundle-generator/blob/0166e27cbd33685837d97ef0962e6b59597fca4c/src/compile-dts.ts#L21) `outDir` compiler option since v3.2.0. Do you use composite projects?

Is it possible to create a repro for that please? With tsconfig and all needed sources (I don't need all of them, just for a minimal repro). I think it might help much to solve the issue.

**Comment by igorpupkinable** [comment]
Thank you for the reply.

Indeed, I use composite project, but it should not affect the matter, because bundle generator is executed with in the package itself.
Please find a cut down repo [here](https://github.com/igorpupkinable/dts-bundle-generator-issues-147).

It works if I generate type definitions upfront with `yarn test:types` and then build with `yarn build`.
You will see an output in `packages/ui-core/lib/`

However, it does not work if `yarn build` from scratch.

**Comment by timocov** [comment]
The reason is `declarationDir` compiler option. This is a bug, I'll fix it shortly. Thanks for reporting!

**Comment by timocov** [comment]
The fix is just published with [5.7.0 version](https://github.com/timocov/dts-bundle-generator/releases/tag/v5.7.0).

**Comment by igorpupkinable** [comment]
It works. Thank you for fixing.
